btk/utils.py

```diff
@@ -1,489 +0,0 @@
-        """
-        # Astropy table with entries corresponding to true sources
-        blend_op, deblend_op, _ = next(self.meas_generator)
-        true_tables = blend_op['blend_list']
-        detected_tables = []
-        for i in range(len(true_tables)):
-            detected_centers = deblend_op[i]['peaks']
-            detected_table = astropy.table.Table(detected_centers,
-                                                 names=['dx', 'dy'])
-            detected_tables.append(detected_table)
-        return true_tables, detected_tables
-
-
-class Stack_metric_params(btk.compute_metrics.Metrics_params):
-    def __init__(self, *args, **kwargs):
-        super(Stack_metric_params, self).__init__(*args, **kwargs)
-        """Class describing functions to return results of
-         detection/deblending/measurement algorithm in meas_generator. Each
-         blend results yielded by the meas_generator for a batch.
-    """
-
-    def get_detections(self):
-        """Returns blend catalog and detection catalog for detection performed
-
-        Returns:
-            Results of the detection algorithm are returned as:
-                true_tables: List of astropy Table of the blend catalogs of the
-                    batch. Length of tables must be the batch size. x and y
-                    coordinate values must be under columns named 'dx' and 'dy'
-                    respectively, in pixels from bottom left corner as (0, 0).
-                detected_tables: List of astropy Table of output from detection
-                    algorithm. Length of tables must be the batch size. x and y
-                    coordinate values must be under columns named 'dx' and 'dy'
-                    respectively, in pixels from bottom left corner as (0, 0).
-        """
-        # Astropy table with entries corresponding to true sources
-        blend_op, _, cat = next(self.meas_generator)
-        true_tables = blend_op['blend_list']
-        detected_tables = []
-        for i in range(len(true_tables)):
-            detected_centers = np.stack(
-                [cat[i]['base_NaiveCentroid_x'],
-                 cat[i]['base_NaiveCentroid_y']],
-                axis=1)
-            detected_table = astropy.table.Table(detected_centers,
-                                                 names=['dx', 'dy'])
-            detected_tables.append(detected_table)
-        return true_tables, detected_tables
-
-
-def get_detection_eff_matrix(summary_table, num):
-    """Computes the detection efficiency table for input detection summary
-    table.
-
-    Input argument num sets the maximum number of true detections for which the
-    detection efficiency matrix is to be created for. Detection efficiency is
-    computed for number of true objects in the range (1-num).
-
-    Args:
-        summary(`numpy.array`) : Detection summary as a table [N, 5].
-        num(int): Maximum number of true objects to create matrix for. Number
```

Comment:
Why set a maximum number of true objects and not use all true objects instead?

Reply:
Hmm I think I understand, but it is not the max number of true objects, but rather the max number of detected objects?